src/Catty/Parser&Serialializer/ProjectMigrator.swift

```diff
@@ -0,0 +1,96 @@
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+class ProjectMigrator: ProjectMigratorProtocol {
+
+    static let minimumCatrobatLanguageVersionForScenes: Float = 0.992
+    var fileManager: CBFileManager
+
+    init(fileManager: CBFileManager) {
+        self.fileManager = fileManager
+    }
+
+    func migrateToScene(project: Project) throws {
+
+        guard let versionNumber = Float(project.header.catrobatLanguageVersion) else {
+            throw ProjectMigratorError.unknown(description: "Unable to convert version number to Float")
+        }
+
+        if versionNumber < ProjectMigrator.minimumCatrobatLanguageVersionForScenes {
+            throw ProjectMigratorError.unsupportedCatrobatLanguageVersion
+        }
+
+        let projectPath = project.projectPath()
+
+        if !fileManager.directoryExists(projectPath) {
+            throw ProjectMigratorError.pathNotFound
+        }
+
+        let directoryPath = "\(projectPath + kLocalizedScene) 1"
+
+        if !fileManager.directoryExists(directoryPath) {
+            fileManager.createDirectory(directoryPath)
+        }
+
+        if fileManager.directoryExists(directoryPath) {
+            self.moveToSceneDirectory(for: kScreenshotAutoFilename, projectPath: projectPath, isFile: true)
+            self.moveToSceneDirectory(for: kProjectImagesDirName, projectPath: projectPath, isFile: false)
+            self.moveToSceneDirectory(for: kProjectSoundsDirName, projectPath: projectPath, isFile: false)
+            self.moveToSceneDirectory(for: kScreenshotManualFilename, projectPath: projectPath, isFile: true)
+            self.moveToSceneDirectory(for: kScreenshotFilename, projectPath: projectPath, isFile: true)
+        } else {
+            throw ProjectMigratorError.unknown(description: "Unable to create \(kLocalizedScene) directory")
+        }
+
+    }
+
+    private func moveToSceneDirectory(for projectResouceName: String, projectPath: String, isFile: Bool) {
+
+        let fileOrDirectoryAtPath = projectPath + projectResouceName
+        let fileOrDirectoryToPath = "\(projectPath + kLocalizedScene) 1/\(projectResouceName)"
+
+        if isFile {
+
+            if !fileManager.fileExists(fileOrDirectoryAtPath) {
+                return
+            }
+
+            fileManager.moveExistingFile(atPath: fileOrDirectoryAtPath, toPath: fileOrDirectoryToPath, overwrite: false)
+        } else {
```

Comment:
Please change to`else if fileManager.directoryExists` to reduce cyclomatic complexity.